### PR TITLE
NETOBSERV-1597: skip RecordKeyMissing error

### PR DESCRIPTION
**Pull request by jpinsonneau**
## Description

Skip all `RecordKeyMissing` and consider these as `0` since we removed empty fields from eBPF agent.

https://github.com/netobserv/netobserv-ebpf-agent/blob/release-1.5/pkg/decode/decode_protobuf.go#L61-L67

If this needs to be configurable for specific cases please let me know. 

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [X] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [X] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: pkg/pipeline/encode/metrics_common.go [hunk 1]
```diff
@@ -228,8 +228,8 @@ func (m *MetricsCommonStruct) extractGenericValue(flow config.GenericMap, info *
 	}
 	val, found := flow[info.ValueKey]
 	if !found {
-		m.errorsCounter.WithLabelValues("RecordKeyMissing", info.Name, info.ValueKey).Inc()
-		return nil
+		// No value means 0 to keep storage lightweight
+		return 0
```
Comment (jotak) [inline pkg/pipeline/encode/metrics_common.go:232]: we should still return nil, I think: for counters, returning 0 or nil will have the same effect (ie. adding nothing to the counter) but for histograms, return 0 would actually corrupt the stats computation

**Review by jotak — APPROVED** [review]

**Comment by jotak** [comment]
thanks @memodi 
/approve

File: pkg/pipeline/encode/metrics_common.go (hunk 1 above)
Comment (jpinsonneau) [reply pkg/pipeline/encode/metrics_common.go]: It depends of the histograms I guess... Do you have some concrete examples in mind ?

For data consistency we should probably add filters before doing such assumptions. For example, DNS metrics must have DNSId, DNSLatencyMs fields etc. That's what we do on plugin query side.

Comment (jotak) [reply pkg/pipeline/encode/metrics_common.go]: All the histograms that we use are latency stuff, so I don't see a case where we would want to say: no latency == 0ms. If a flow is missing RTT, or Dns latency, then we should just ignore it for histogram stats. Expecting an explicit filter would add extra processing and could easily be error-prone IMO. I'd return your question: do you have an example in mind where we need to say nil == 0 for an histogram ?

Comment (jpinsonneau) [reply pkg/pipeline/encode/metrics_common.go]: bytes, packets and their respective drops fields must be considered as 0 when keys are not present

Comment (jotak) [reply pkg/pipeline/encode/metrics_common.go]: yes but that's fine these are counters not histograms, adding 0 to the counter is equivalent to ignoring it

**Pull request merged**